Kentico.Kontent.Management/Modules/ActionInvoker/ImageTransformationConverter.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using Kentico.Kontent.Management.Models.AssetRenditions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Kentico.Kontent.Management.Modules.ActionInvoker;
+
+internal class ImageTransformationConverter : JsonConverter
+{
+    private static readonly JsonSerializerSettings _specifiedSubclassConversion = new JsonSerializerSettings() { ContractResolver = new BaseSpecifiedConcreteClassConverter() };
+
+    public override bool CanConvert(Type objectType) => objectType == typeof(ImageTransformation);
+
+    public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+    {
+        var jObject = JObject.Load(reader);
+        var type = jObject["mode"].ToObject<ImageTransformationMode>();
+
+        return type switch
+        {
+            ImageTransformationMode.Rect =>
+                JsonConvert.DeserializeObject<RectangleResizeTransformation>(jObject.ToString(), _specifiedSubclassConversion),
+            _ => throw new NotImplementedException()
```

Comment:
Could we throw `InvalidEnumArgumentException` with some descriptive message? Or do you prefer any other exception?

Reply:
Yeah, that's much better :) I hope I used the ctor correctly.